Assets/Scripts/EnergyBlast.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering.Universal.Internal;
+
+public class EnergyBlast : MonoBehaviour
+{
+    public RectTransform reticle;
+
+    public float maxRange;
+    public int maxNumOfCharges;
+    public float rechargeTimer;
+    public float timeBetweenShots;
+
+    private int currNumOfCharges;
+    private float currRechargeTimer = 0f;
+    private float currTimeBetweenShots;
+
+    private void Start()
+    {
+        currNumOfCharges = maxNumOfCharges;
+        currTimeBetweenShots = timeBetweenShots;
+    }
+
+    private void Update()
+    {
+        Recharge();
+
+        if (currTimeBetweenShots < timeBetweenShots)
+        {
+            currTimeBetweenShots += Time.deltaTime;
+        }
+
+        if (Input.GetMouseButtonDown(0) && currTimeBetweenShots >= timeBetweenShots && currNumOfCharges > 0)
```

Comment:
Implement into player inputs script
- [ ] add the keycode parameter
- [ ] seperate the shooting and player input shooting